### ISSUE TITLE
Custom formatter filePath is absolute. Docs say they are relative to working dir.

### ISSUE THREAD

**Issue by ViggoV** [eslint/eslint#11156]
**Tell us about your environment**
Webpack 4.x, React 16.x

* **ESLint Version:** 5.9.0
* **Node Version:** 8.11.4
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
`babel-eslint` (for another day; the local install requires `babel-eslint` or throws `unexpected token =`)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true,
    "mocha": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 2018,
    "ecmaFeatures": {
      "JSX": true,
      "modules": true
    }
  },
  "rules": {
    "indent": "off",
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ],
    "no-console": [
      "warn"
    ],
    "multiline-comment-style": [
      "warn",
      "separate-lines"
    ]
  },
  "plugins": [
    "react"
  ]
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I am trying to create a custom formatter to display a summary of the output. The documentation says right [here](https://eslint.org/docs/developer-guide/working-with-custom-formatters) that the `filePath`'s included in the results are supposed to be relative to the working directory. However in my case they are absolute, in relation to the system root. Had they been relative to my home directory I would have suspected that my global eslint instance was called, but that doesn't seem to be the case.

It can be reproduced as simple as...
```js
module.exports = (results) => {
  return results.map(r => r.filePath);
};
```
Which in my case prints:

```
[ '/Users/vef/Documents/projects/code/key-figures-module/src/App/index.jsx',
  '/Users/vef/Documents/projects/code/key-figures-module/src/App/Main/Body/index.jsx',
  '/Users/vef/Documents/projects/code/key-figures-module/src/App/Main/Header/index.jsx',
  '/Users/vef/Documents/projects/code/key-figures-module/src/App/Main/index.jsx',
  '/Users/vef/Documents/projects/code/key-figures-module/src/App/Sidebar/index.jsx',
  '/Users/vef/Documents/projects/code/key-figures-module/src/index.jsx' ]
```

I run eslint via an npm script, which looks like this
```json
{
  "lint": "eslint src/. --ext .jsx --ext .js --color -f ./node_modules/eslint-overview-formatter; exit 0",
}
```

**What did you expect to happen?**
I expected the `filePath`'s to be relative to my working directory. In other words the above output should be closer to
```
[ './src/App/index.jsx',
  './src/App/Main/Body/index.jsx',
  './src/App/Main/Header/index.jsx',
  './src/App/Main/index.jsx',
  './src/App/Sidebar/index.jsx',
  './src/index.jsx' ]
```

**What actually happened? Please include the actual, raw output from ESLint.**
The actual raw output is presented above.

**Are you willing to submit a pull request to fix this bug?**
If it makes sense, yes. But I don't have one at hand.

**Comment by ViggoV** [comment]
I came up with a workaround using `path`:
```js
const path = require('path');
module.exports = (results) => {
  let root = path.resolve('.') // get absolute path of project root
  let fixed = results.map(r => {
    // set relativePath to difference between root and filePath 
    return {...r, relativePath: path.relative(root, r.filePath)})
  };
  return results.map(r => r.filePath);
};
```

Output:
```
[ 'src/App/index.jsx',
  'src/App/Main/Body/index.jsx',
  'src/App/Main/Header/index.jsx',
  'src/App/Main/index.jsx',
  'src/App/Sidebar/index.jsx',
  'src/index.jsx' ]
```

**Comment by g-plane** [comment]
I think this can be closed since https://github.com/eslint/eslint/commit/9c31625f19176664ef76dcf088ce50703c41c324 fixed it.
